compiler/transforms/snax_copy_to_dma.py

```diff
@@ -0,0 +1,61 @@
+from xdsl.dialects import builtin, func, memref, arith
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriteWalker,
+    PatternRewriter,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.traits import SymbolTable
+
+
+class InsertFunctionCalls(RewritePattern):
+    """
+    Looks for memref copy operations and insert a snitch 1d dma call
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: memref.CopyOp, rewriter: PatternRewriter):
+        # Exctract size information
+        zero_const = arith.Constant.from_int_and_width(0, builtin.IndexType())
+        dim_op = memref.Dim.from_source_and_index(op.source, zero_const.result)
+
+        # Extract source and destination pointers
+        source_ptr_op = memref.ExtractAlignedPointerAsIndexOp.get(op.source)
+        dest_ptr_op = memref.ExtractAlignedPointerAsIndexOp.get(op.destination)
+
+        # Make function call
+        func_call = func.Call(
+            "snax_dma_1d_transfer",
+            [source_ptr_op.aligned_pointer, dest_ptr_op.aligned_pointer, dim_op.result],
+            [],
+        )
+
+        # Replace op with function call
+        rewriter.insert_op_before_matched_op(
+            [zero_const, dim_op, source_ptr_op, dest_ptr_op]
+        )
+        rewriter.replace_op(op, func_call)
```

Comment:
Wondering whether we should check here if the memref.copy is actually copying a 1D memref?
Otherwise some reshaping is necessary?

Reply:
Yes, reshaping is not supported yet, so I just included a check that both memrefs are of rank 1